src/terrain/terrain.cpp

```diff
@@ -188,13 +178,12 @@ Terrain::init_area(int area_x, int area_y, terrain_generation::LandGenerator gen
 
 void
 Terrain::init_chunks() {
-    for (int xyz = 0;
-         xyz < ((X_MAX - 1) / Chunk::size + 1) * ((Y_MAX - 1) / Chunk::size + 1)
-                   * ((Z_MAX - 1) / Chunk::size + 1);
-         xyz += 1) {
-        auto [x, y, z] =
-            sop(xyz, ((X_MAX - 1) / Chunk::size + 1), ((Y_MAX - 1) / Chunk::size + 1),
-                ((Z_MAX - 1) / Chunk::size + 1));
+    // chunk length in _ direction
+    uint32_t C_length_X = ((X_MAX - 1) / Chunk::size + 1);
+    uint32_t C_length_Y = ((Y_MAX - 1) / Chunk::size + 1);
+    uint32_t C_length_Z = ((Z_MAX - 1) / Chunk::size + 1);
+    for (size_t xyz = 0; xyz < C_length_X * C_length_Y * C_length_Z; xyz += 1) {
+        auto [x, y, z] = sop(xyz, C_length_X, C_length_Y, C_length_Z);
```

Comment:
What does sop stand for

Reply:
sop is the reverse of pos which is the position of a tile in the `tiles` vector